src/community/backup-restore/rest/src/main/java/org/geoserver/backuprestore/rest/RestoreResource.java

```diff
@@ -0,0 +1,218 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2016 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.rest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.backuprestore.RestoreExecutionAdapter;
+import org.geoserver.backuprestore.rest.format.RestoreJSONReader;
+import org.geoserver.backuprestore.rest.format.RestoreJSONWriter;
+import org.geoserver.rest.PageInfo;
+import org.geoserver.rest.RestletException;
+import org.geoserver.rest.format.DataFormat;
+import org.geoserver.rest.format.StreamDataFormat;
+import org.restlet.data.Form;
+import org.restlet.data.MediaType;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+import org.restlet.data.Status;
+
+/**
+ * REST resource for 
+ * 
+ * <pre>/br/restore[/&lt;restoreId&gt;]?expand=1</pre>
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public class RestoreResource  extends BaseResource {
+
+    public RestoreResource(Backup backupFacade) {
+        super(backupFacade);
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    @Override
+    protected List<DataFormat> createSupportedFormats(Request arg0, Response arg1) {
+        return (List) Arrays.asList(new RestoreJSONFormat(MediaType.APPLICATION_JSON),
+                new RestoreJSONFormat(MediaType.TEXT_HTML));
+    }
+
+    @Override
+    public boolean allowPost() {
+        return true;
+    }
+    
+    @Override
+    public boolean allowPut() {
+        return false;
+    }
+    
+    @Override
+    public void handleGet() {
+        DataFormat formatGet = getFormatGet();
+        if (formatGet == null) {
+            formatGet = new RestoreJSONFormat(MediaType.APPLICATION_JSON);
+        }
+        Object lookupContext = lookupContext(true, false);
+        if (lookupContext == null) {
+            // this means a specific lookup failed
+            getResponse().setStatus(Status.CLIENT_ERROR_NOT_FOUND);
+        } else {
+            getResponse().setEntity(formatGet.toRepresentation(lookupContext));
+        }
+    }
+    
+    @Override
+    public void handlePost() {
+        Object obj = lookupContext(true, true);
```

Comment:
Is there a way for someone to point to a local zip file instead of uploading it? The file could be quite big if we are in the vicinity of tens of thousands of layers.